src/main/java/org/openhab/tools/analysis/checkstyle/BuildPropertiesExternalLibrariesCheck.java

```diff
@@ -0,0 +1,124 @@
+/**
+ * Copyright (c) 2010-2017 by the respective copyright holders.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.tools.analysis.checkstyle;
+
+import static org.openhab.tools.analysis.checkstyle.api.CheckConstants.PROPERTIES_EXTENSION;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import org.eclipse.pde.core.build.IBuildEntry;
+import org.openhab.tools.analysis.checkstyle.api.AbstractExternalLibrariesCheck;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.puppycrawl.tools.checkstyle.api.CheckstyleException;
+
+/**
+ * A check that asserts that all jar files in the lib folder are added to build.properties and all jar files in
+ * build.properties
+ * are added to the lib folder.
+ *
+ * @author Velin Yordanov - Initial contribution
+ *
+ */
+public class BuildPropertiesExternalLibrariesCheck extends AbstractExternalLibrariesCheck {
+    private final String BIN_INCLUDES = "bin.includes";
+    private final String FILES_NEED_TO_BE_IN_A_LIB_FOLDER = "All jar files need to be placed inside a lib folder.";
+    private final String JAR_PRESENT_IN_BUILD_PROPERTIES_NOT_IN_LIB = "The file %s is present in the build properties but not in the lib folder.";
+    private final String JAR_PRESENT_IN_LIB_NOT_IN_BUILD_PROPERTIES = "The jar file %s is present in the lib folder but is not present in the build properties";
+    private final Logger logger = LoggerFactory.getLogger(BuildPropertiesExternalLibrariesCheck.class);
+
+    public BuildPropertiesExternalLibrariesCheck() {
+        // build.properties will not be explicitly processed by the check.
+        setFileExtensions(PROPERTIES_EXTENSION);
+    }
+
+    @Override
+    protected void processFiltered(File file, List<String> lines) throws CheckstyleException {
```

Comment:
This method has been changed in Checkstyle 8.1, please update - https://github.com/openhab/static-code-analysis/commit/5ae6ff49c48a5941ded214e1fe6f6a0af8452686